tests/integration/test_charm.py

```diff
@@ -17,14 +18,38 @@
 METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
 APP_NAME = METADATA["name"]
 
+METADATA_SERVER = yaml.safe_load(Path("../temporal-k8s-operator/metadata.yaml").read_text())
+APP_NAME_SERVER = METADATA_SERVER["name"]
 
-@pytest.fixture(name="deploy", scope="module")
+
+@pytest_asyncio.fixture(name="deploy", scope="module")
 async def deploy(ops_test: OpsTest):
+    """The app is up and running."""
     charm = await ops_test.build_charm(".")
-    resources = {"temporal-admin-image": METADATA["resources"]["temporal-admin-image"]["upstream-source"]}
+    resources = {"temporal-admin-image": METADATA["containers"]["temporal-admin"]["upstream-source"]}
```

Comment:
So it is in fact possible to run integration tests with a local image. I do that in the XYZ charm some links are below if you would like to see how that's done.
1. The Github [action](https://github.com/canonical/xyz/blob/5a5fb6848eca7bd54e344b763d8e8d2512fd3d33/.github/workflows/ci.yaml#L48) to push the images to local microk8s registry
2. The actual [test](https://github.com/canonical/xyz/blob/main/charm/tests/integration/conftest.py#L16) referencing the local image.